pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -575,204 +576,209 @@ func (c *Check) StatementMetrics() (int, error) {
-								}
-								if stepRow.Cardinality.Valid {
-									stepPayload.Cardinality = stepRow.Cardinality.Float64
-								}
-								if stepRow.Bytes.Valid {
-									stepPayload.Bytes = stepRow.Bytes.Float64
-								}
-								if stepRow.PartitionStart.Valid {
-									stepPayload.PartitionStart = stepRow.PartitionStart.String
-								}
-								if stepRow.PartitionStop.Valid {
-									stepPayload.PartitionStop = stepRow.PartitionStop.String
-								}
-								if stepRow.CPUCost.Valid {
-									stepPayload.CPUCost = stepRow.CPUCost.Float64
-								}
-								if stepRow.IOCost.Valid {
-									stepPayload.IOCost = stepRow.IOCost.Float64
-								}
-								if stepRow.TempSpace.Valid {
-									stepPayload.TempSpace = stepRow.TempSpace.Float64
-								}
-								handlePredicate("access", stepRow.AccessPredicates, &stepPayload.AccessPredicates, statementMetricRow, c, o)
-								handlePredicate("filter", stepRow.FilterPredicates, &stepPayload.FilterPredicates, statementMetricRow, c, o)
-								if stepRow.Projection.Valid {
-									stepPayload.Projection = stepRow.Projection.String
-								}
-								if stepRow.LastStarts != nil {
-									stepPayload.LastStarts = *stepRow.LastStarts
-								}
-								if stepRow.LastOutputRows != nil {
-									stepPayload.LastOutputRows = *stepRow.LastOutputRows
-								}
-								if stepRow.LastCRBufferGets != nil {
-									stepPayload.LastCRBufferGets = *stepRow.LastCRBufferGets
-								}
-								if stepRow.LastDiskReads != nil {
-									stepPayload.LastDiskReads = *stepRow.LastDiskReads
-								}
-								if stepRow.LastDiskWrites != nil {
-									stepPayload.LastDiskWrites = *stepRow.LastDiskWrites
-								}
-								if stepRow.LastElapsedTime != nil {
-									stepPayload.LastElapsedTime = *stepRow.LastElapsedTime
-								}
-								if stepRow.LastMemoryUsed != nil {
-									stepPayload.LastMemoryUsed = *stepRow.LastMemoryUsed
-								}
-								if stepRow.LastDegree != nil {
-									stepPayload.LastDegree = *stepRow.LastDegree
-								}
-								if stepRow.LastTempsegSize != nil {
-									stepPayload.LastTempsegSize = *stepRow.LastTempsegSize
-								}
-								if stepRow.PlanCreated.Valid && stepRow.PlanCreated.String != "" {
-									oraclePlan.Timestamp = stepRow.PlanCreated.String
-								}
-								if stepRow.OptimizerMode.Valid && stepRow.OptimizerMode.String != "" {
-									oraclePlan.OptimizerMode = stepRow.OptimizerMode.String
-								}
-								if stepRow.Other.Valid && stepRow.Other.String != "" {
-									oraclePlan.Other = stepRow.Other.String
-								}
-
-								oraclePlan.PDBName = statementMetricRow.PDBName
-								oraclePlan.SQLID = statementMetricRow.SQLID
-								oraclePlan.ForceMatchingSignature = statementMetricRow.ForceMatchingSignature
-								oraclePlan.Executions = statementMetricRow.Executions
-								oraclePlan.ElapsedTime = statementMetricRow.ElapsedTime
+				if c.config.ExecutionPlans.Enabled {
```

Comment:
When a  FQT cache miss happens, I think the bottleneck is fetching the execution plan. Is it worth to always send FQT but only skip plan? Correct me if it's a bad idea and would lead to incomplete data or broken user experience

Reply:
Fetching the execution plan doesn't seem to be the bottleneck:

![image](https://github.com/DataDog/datadog-agent/assets/18366081/eb4d2d04-3263-4bb5-b271-6abead71877b)

It took only 1s second to fetch ~100 plans. Hence, out of 25s elapsed time, only 12s was spent in the database. I suspect, the rest of the time was spent emitting the events, possibly due to network latency. If that's the case, it would be correct to to stop emitting also the FQT events.

Reply:
If we were to stop sending FQT and plan, will it make more sense to put the check `time.Since(start).Seconds() >= float64(c.config.QueryMetrics.MaxRunTime)` towards the start of the loop `for i, statementMetricRow := range statementMetricsAll`, and `break` the loop when ` time since start` exceed `MaxRunTime`? 
Part of that is also because `sendFqtAndPlan` won't be reset to `true` within the loop once it's set to `false`.

Reply:
We can't break the loop because the loop also builds the query metrics payload which we would like to complete. In other words, once `sendFqtAndPlan` is set to `false` we would like to continue processing the query metrics payload, which is by orders of magnitude faster than emitting FQT and plan events, and just stop sending the FQT and plan events.

Reply:
Upon closer examination, I discovered that the time isn't consumed by event emission. I need to conduct further measurements to identify the source of the time spent that is unrelated to database operations or CPU usage.